### ISSUE TITLE
[UI] The Jobs page shows "No jobs found" after being idle for two minutes.

### ISSUE THREAD

**Issue by kodecraft-pau** [hashicorp/nomad#23228]
### Nomad version
Nomad v1.8.0
BuildDate 2024-05-28T17:38:17Z
Revision 28b82e4b2259fae5a62e2ed47395334bea5a24c4

### Operating system and Environment details
Distributor ID: Ubuntu
Description:    Ubuntu 24.04 LTS
Release:        24.04
Codename:       noble

### Issue
The Jobs page shows "No jobs found" after being idle for a few minutes. This might worry users if they have many jobs running and become lost (even if only in display). It returns to normal once you refresh the page.

### Reproduction steps
1. Upgrade Nomad Version from 1.7.5 to 1.8.0 _(not sure if this helps)_
2. Login to Nomad and navigate to Jobs menu
3. Leave the Page/Tab or become idle for 2 minutes

#### Expected Result
The page should display the jobs whether you are idle or not.
![image](https://github.com/hashicorp/nomad/assets/155626785/443df980-c994-4b7d-acfa-dd03cf15a6d9)

#### Actual Result
The Jobs page shows "No jobs found." when you become idle for two minutes.
![image](https://github.com/hashicorp/nomad/assets/155626785/a77176df-5d82-49ce-af22-06176858193d)

![image](https://github.com/hashicorp/nomad/assets/155626785/7163a03f-34b2-48bd-b0d0-dd48a48d5ebe)

**Comment by erwin-willems** [comment]
I have this behavior too in my clusters

**Comment by faryon93** [comment]
Seeing the same behaviour in my cluster after updating to 1.8.0

It looks like the web-ui tries to access some endpoints in the nomad api which have "long running" semantics. I think the is the new "live update" of the jobs page. These should be the requests with a `index` parameter.

![Screenshot_20240614_213841](https://github.com/hashicorp/nomad/assets/12020944/041e3c4d-1d54-41b1-8ddd-1a91451b8622)

I run a reverse proxy in front of nomad ui which has a upstream timeout of 60s. The first request is aborted exactly after 60s (see chrome dev tools screenshot). You can tell from the access log that nginx aborted the request, because there was no content sent by nomad.

```
2024/06/14 19:32:57 [warn] 99#99: *37748 upstream server temporarily disabled while reading response header from upstream, client: 10.0.28.139, server: nomad-ui, request: "GET /v1/jobs/statuses?filter=&namespace=%2A&per_page=25&index=1209611 HTTP/2.0", upstream: "http://192.168.70.102:4646/v1/jobs/statuses?filter=&namespace=%2A&per_page=25&index=1209611", host: "nomad-ui", referrer: "https://nomad-ui/ui/jobs"
2024/06/14 19:32:57 [error] 99#99: *37748 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 10.0.28.139, server: nomad-ui, request: "GET /v1/jobs/statuses?filter=&namespace=%2A&per_page=25&index=1209611 HTTP/2.0", upstream: "http://192.168.70.102:4646/v1/jobs/statuses?filter=&namespace=%2A&per_page=25&index=1209611", host: "nomad-ui, referrer: "https://nomad-ui/ui/jobs"
```

Nginx temporarily disables the timed-out upstream so the retries by the webfrontend fail as well until the frontend gives up and no further retries are made.

In my case I should have read the documentation a little better because it is clearly state by the docs:

https://developer.hashicorp.com/nomad/tutorials/manage-clusters/reverse-proxy-ui#extend-connection-timeout

**Comment by philrenaud** [comment]
Thanks everybody for your reports and patience on this issue — we've fixed the bug in #23427  and given options to pause fetching in case lengthening proxy timeout is not immediately possible.
